test/blas1_test.hpp

```diff
@@ -0,0 +1,207 @@
+    return N;
+  }
+
+  // getting the stride in the same way as the size above
+  size_t test_strd() {
+    static bool first = true;
+    static size_t N;
+    if (first) {
+      first = false;
+      N = ((rand() & 1) * (rand() % 5)) + 1;
+    }
+    return N;
+  }
+
+  template <typename DataType,
+            typename value_type = typename DataType::value_type>
+  static void set_rand(DataType &vec, size_t _N) {
+    value_type left(-1), right(1);
+    for (size_t i = 0; i < _N; ++i) {
+      vec[i] = value_type(rand() % int(right - left) * 1000) * .001 - right;
+    }
+  }
+
+  template <typename DataType,
+            typename value_type = typename DataType::value_type>
+  static void print_cont(const DataType &vec, size_t _N,
+                         std::string name = "vector") {
+    std::cout << name << ": ";
+    for (size_t i = 0; i < _N; ++i) std::cout << vec[i] << " ";
+    std::cout << std::endl;
+  }
+
+  template <typename DataType,
+            typename value_type = typename DataType::value_type>
+  static cl::sycl::buffer<value_type, 1> make_buffer(DataType &vec) {
+    return cl::sycl::buffer<value_type, 1>(vec.data(), vec.size());
+  }
+
+  template <typename value_type>
+  static vector_view<value_type, cl::sycl::buffer<value_type>> make_vview(
+      cl::sycl::buffer<value_type, 1> &buf) {
+    return vector_view<value_type, cl::sycl::buffer<value_type>>(buf);
+  }
+
+  template <typename DeviceSelector,
+            typename = typename std::enable_if<
+                std::is_same<ExecutorType, SYCL>::value>::type>
+  static cl::sycl::queue make_queue(DeviceSelector s) {
+    return cl::sycl::queue(s, [=](cl::sycl::exception_list eL) {
+      try {
+        for (auto &e : eL) {
+          std::rethrow_exception(e);
+        }
+      } catch (cl::sycl::exception &e) {
+        std::cout << "E " << e.what() << std::endl;
+      } catch (std::exception &e) {
+        std::cout << "Standard Exception " << e.what() << std::endl;
+      } catch (...) {
+        std::cout << " An exception " << std::endl;
+      }
+    });
+  }
+};
+
+// unpacking the parameters within the test function
+// B is blas_templ_struct
+// TestClass is BLAS1_Test<B>
+// T is default (scalar) type for the test (e.g. float, double)
+// C is the container type for the test (e.g. std::vector)
+// E is the executor kind for the test (sequential, openmp, sycl)
+#define B1_TEST(name) TYPED_TEST(BLAS1_Test, name)
+#define UNPACK_PARAM(test_name)                        \
+  using ScalarT = typename TypeParam::scalar_t;        \
+  using TestClass = BLAS1_Test<TypeParam>;             \
+  using ExecutorType = typename TypeParam::executor_t; \
+  using test = class test_name;
+// TEST_SIZE determines the size based on the suggestion
+#define TEST_SIZE                                                       \
+  ((option_size<test>::value == ::RANDOM_SIZE) ? TestClass::test_size() \
+                                               : option_size<test>::value)
```

Comment:
why can't this be simply a function that returns a boolean instead of a macro? it is much easier to debug